roles/matrix-bot-postmoogle/templates/systemd/matrix-bot-postmoogle.service.j2

```diff
@@ -0,0 +1,40 @@
+#jinja2: lstrip_blocks: "True"
+[Unit]
+Description=Matrix helpdesk bot
+{% for service in matrix_bot_postmoogle_systemd_required_services_list %}
+Requires={{ service }}
+After={{ service }}
+{% endfor %}
+{% for service in matrix_bot_postmoogle_systemd_wanted_services_list %}
+Wants={{ service }}
+{% endfor %}
+DefaultDependencies=no
+
+[Service]
+Type=simple
+Environment="HOME={{ matrix_systemd_unit_home_path }}"
+ExecStartPre=-{{ matrix_host_command_sh }} -c '{{ matrix_host_command_docker }} kill matrix-bot-postmoogle 2>/dev/null || true'
+ExecStartPre=-{{ matrix_host_command_sh }} -c '{{ matrix_host_command_docker }} rm matrix-bot-postmoogle 2>/dev/null || true'
+
+ExecStart={{ matrix_host_command_docker }} run --rm --name matrix-bot-postmoogle \
+			--log-driver=none \
+			--user={{ matrix_user_uid }}:{{ matrix_user_gid }} \
+			--cap-drop=ALL \
+			--read-only \
+			--network={{ matrix_docker_network }} \
+			--env-file={{ matrix_bot_postmoogle_config_path }}/env \
+			-p {{ matrix_bot_postmoogle_port }}:{{ matrix_bot_postmoogle_port }} \
```

Comment:
Isn't this (with the default configuration) trying to bind to port `25` inside the container?

Since it's a [privileged port](https://www.w3.org/Daemon/User/Installation/PrivilegedPorts.html) (` < 1024`), this is supposed to require special permissions. We either need to run the container as `root` (not great) or add `--cap-add=NET_ADMIN` (also not great).

Alternatively, we can make the ports different. We can make Postmoogle (in the container) bind to a non-privileged port (e.g. `2525`) and then have Docker itself do the forwarding (`-p 25:2525`).

----

The `email2matrix` bridge hardcodes port `2525` for the container program and only ever changes the exposed port, controlled by a `matrix_email2matrix_smtp_host_bind_port` setting.

https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/5fafbab7ae4b6fd090c152144932da6e57f883cd/roles/matrix-email2matrix/templates/config.json.j2#L4

https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/5fafbab7ae4b6fd090c152144932da6e57f883cd/roles/matrix-email2matrix/templates/systemd/matrix-email2matrix.service.j2#L20

https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/5fafbab7ae4b6fd090c152144932da6e57f883cd/roles/matrix-email2matrix/defaults/main.yml#L25-L30

Reply:
> Isn't this (with the default configuration) trying to bind to port 25 inside the container?

Yes, but you can do that inside the container, [even with custom user `postmoogle`](https://gitlab.com/etke.cc/postmoogle/-/blob/main/Dockerfile#L17). It works with the 25 port without additional capability